### PR TITLE
[raz][abfs] Improve url params encoding for RAZ ABFS client

### PR DESCRIPTION
**Pull request by Harshg999**
## What changes were proposed in this pull request?

- Earlier we were encoding the url params only for small set of use-cases but it turns out we should always encode the params to cover all use-cases.
- To support the above observation, I'd to override the _make_url method of `HttpClient` class and change its `urlencode` method to use `quote()` method instead of default `quote_plus()`. This was required to fix scenarios of whitespace present in the path which regressed after the above change.

## How was this patch tested?

- Tested manually.
- Updating existing unit tests.
- Adding new unit tests.

### CODE REVIEW

File: desktop/core/src/desktop/lib/rest/raz_http_client.py [hunk 1]
```diff
@@ -48,12 +46,7 @@ def execute(self, http_method, path, params=None, data=None, headers=None, allow
     https://{storageaccountname}.dfs.core.windows.net/{container}/{path}?sv=2014-02-14&sr=b&
     sig=pJL%2FWyed41tptiwBM5ymYre4qF8wzrO05tS5MCjkutc%3D&st=2015-01-02T01%3A40%3A51Z&se=2015-01-02T02%3A00%3A51Z&sp=r
     """
-    if sys.version_info[0] < 3 and isinstance(path, unicode):
-      path = path.encode('utf-8')
-
-    do_urlencode = True if params and 'directory' in params and '%' in params['directory'] else False
-
-    url = self._make_url(lib_urlquote(path), params, do_urlencode=do_urlencode)
+    url = self._make_url(quote(path), params)
```
Comment (bjornalm) [inline desktop/core/src/desktop/lib/rest/raz_http_client.py:49]: How is this URL used later? Is it sent back to the UI or called by the backend?

File: desktop/core/src/desktop/lib/rest/raz_http_client.py [hunk 2]
```diff
@@ -116,3 +109,14 @@ def get_sas_token(self, http_method, username, url, params=None, headers=None):
       return response.get('token')
     else:
       raise PopupException(_('No SAS token in RAZ response'), error_code=403)
+
+  def _make_url(self, path, params, do_urlencode=True):
```
Comment (amitsrivastava) [inline desktop/core/src/desktop/lib/rest/raz_http_client.py:113]: Can you add a comment here to mentioning why you needed to reimplement this function?

**Review by JohanAhlen — APPROVED** [review]
Nice one!

File: desktop/core/src/desktop/lib/rest/raz_http_client.py (hunk 2 above)
Comment (Harshg999) [reply desktop/core/src/desktop/lib/rest/raz_http_client.py:146]: Done.

**Review by agl29 — APPROVED** [review]
Nice unit tests!

**Pull request merged**